src/Engine/GameInput/SelectLaunchable.java

```diff
@@ -0,0 +1,50 @@
+package Engine.GameInput;
+
+import java.util.ArrayList;
+
+import Units.Unit;
+
+/************************************************************
+ * State to choose which Unit will be kicked off the bus.   *
+ ************************************************************/
+class SelectLaunchable extends GameInputState<Unit>
+{
+  public SelectLaunchable(StateData data)
+  {
+    super(data);
+    myStateData.unitLauncher = myStateData.unitActor;
+  }
+
+  @Override
+  protected OptionSet initOptions()
+  {
+    // Collect the units held by unitActor.
+    ArrayList<Unit> launchables = new ArrayList<Unit>();
+    for( Unit cargo : myStateData.unitActor.heldUnits )
+    {
+      if( !cargo.isTurnOver )
+        launchables.add(cargo);
+    }
+    Unit[] cargoArray = new Unit[launchables.size()];
+    launchables.toArray(cargoArray);
+    return new OptionSet(cargoArray);
+  }
+
+  @Override
+  public GameInputState<?> select(Unit option)
+  {
+    GameInputState<?> next = this;
+
+    myStateData.unitActor = option;
```

Comment:
Should we confirm that `option` is actually on this transport before assuming all is well? Probably not critical, but c.f. `SelectCargo`, which does this.

Reply:
If I recall correctly, I was the one who wrote that logic initially, and I did it to avoid casting `Object` to `Unit` rather than any desire to validate that it's actually in the transport.
IMO it's not that useful to validate everywhere: at this point, it won't get here without being valid... and if it goes through here invalid... it will still fail safely when the action/event gets generated.

Reply:
May wanna check that memory (`git blame` can help), and also check `SelectMoveLocation` that does a similar check despite not needing to cast.
And you are right, it's probably not critical since the UI enforces particular behavior. I guess I tend to err on the side of caution, because simple checks cost almost nothing, and fast failing is better than slow failing or erroneously appearing to work.

Reply:
Pretty sure the GameInput rework happened while we were both at the parents' house and we passed code back and forth a bit. `blame` wouldn't know that, and I'm not sure it copes with squash commits either.
I guess it's also possible you wrote it and I generated that reasoning during review.

Works for me.

Reply:
Yeah, maybe. It may also be necessary since we want to validate UI input wherever it's input, but we also want to check action/event validity since the AI doesn't use the UI to build them. I don't want to break the existing working paradigm without properly thinking through the consequences. It may not be as elegant as it could be, but it works.